lib/segment/src/segment_constructor/segment_constructor_base.rs

```diff
@@ -782,7 +785,14 @@ pub fn build_segment(
 
     std::fs::create_dir_all(&segment_path)?;
 
-    let segment = create_segment(None, None, &segment_path, config, &AtomicBool::new(false))?;
+    let segment = create_segment(
+        None,
+        None,
+        &segment_path,
+        config,
+        &AtomicBool::new(false),
+        true,
+    )?;
```

Comment:
Context # 2: This is the main reason for `create_if_missing`. When loading a segment, we want to fail if the index's backing storage is different. But when building a new segment, we want to create the files so they can be populated from scratch.

@timvisee can you confirm this is correct?